README.md

```diff
@@ -108,85 +156,242 @@ in the case of lists sets and scalars, rows with key and value text columns
 
-The tests for 9.2 and later assume that you have access to a redis server
-on the localmachine with no password, and uses database 15, which must be empty,
-and that the redis-cli program is in the PATH when it is run.
-The test script checks that the database is empty before it tries to
+- Redis cursors have some significant limitations. The Redis docs say:
+
+    *A given element may be returned multiple times. It is up to the
+    application to handle the case of duplicated elements, for example only
+    using the returned elements in order to perform operations that are safe
+    when re-applied multiple times*.
+
+  The FDW makes no attempt to detect this situation. Users should be aware of
+  the possibility.
+
+- There was no such thing as a cursor in Redis 2.8- in the SQL sense. Redis
+  releases prior to 2.8 are maintained on the REL9_x_STABLE_pre2.8 branches.
+
+Tests
+-----
+
+The tests for PostgreSQL assume that you have access to a Redis server
+on the local machine with no password, and uses PostgreSQL 15 server with
+*english* locale. This database must be empty, and that the `redis-cli` program
+is in the `PATH` envireonment variable when tests is run.
+The [test](test) script checks that the database is empty before it tries to
 populate it, and it cleans up afterwards.
 
+Some tests as `psql` expected outputs can be found in [test/expected](test/expected) directory.
+ 
+Contributing
+------------
 
-Authors
--------
+Opening issues and pull requests on GitHub are welcome.
+
+Useful links
+------------
+
+### Redis selected documentation
+
+- https://redis.io/commands/
+- https://redis.io/docs/
+- https://redis.io/docs/data-types/
+- https://github.com/redis/hiredis/blob/master/README.md
+
+### Source code
+ 
+- https://github.com/redis/hiredis - hiredis C client library
+- https://github.com/redis/redis - redis DB
+- https://bitbucket.org/adunstan/redis_wrapper/src/master/ - PostgreSQL extension (not FDW) for Redis (also written by Andrew Dunstan)
+- https://github.com/jeffreydwalter/redis_cluster_fdw - Other FDW for Redis
+
+ Reference FDW implementation, `postgres_fdw`
+ - https://git.postgresql.org/gitweb/?p=postgresql.git;a=tree;f=contrib/postgres_fdw;hb=HEAD 
 
-Dave Page
-dpage@pgadmin.org
+### General FDW Documentation
+
+ - https://www.postgresql.org/docs/current/ddl-foreign-data.html
+ - https://www.postgresql.org/docs/current/sql-createforeigndatawrapper.html
+ - https://www.postgresql.org/docs/current/sql-createforeigntable.html
+ - https://www.postgresql.org/docs/current/sql-importforeignschema.html
+ - https://www.postgresql.org/docs/current/fdwhandler.html
+ - https://www.postgresql.org/docs/current/postgres-fdw.html
+
+### Other FDWs
+
+ - https://wiki.postgresql.org/wiki/Fdw
+ - https://pgxn.org/tag/fdw/
+
+License and authors
+-------
+* Dave Page dpage@pgadmin.org
+* Andrew Dunstan andrew@dunslane.net
 
-Andrew Dunstan
-andrew@dunslane.net
+There is no licese file here.
```

Comment:
Please remove this line.

Reply:
Please verify, changed in https://github.com/pg-redis-fdw/redis_fdw/pull/37/commits/90bac01fa56309fe174b8dc94dbcb1abcbb4200b